src/test/txvalidationcache_tests.cpp

```diff
@@ -77,7 +77,9 @@ BOOST_FIXTURE_TEST_CASE(tx_mempool_block_doublespend, Dersig100Setup)
         LOCK(cs_main);
         BOOST_CHECK(m_node.chainman->ActiveChain().Tip()->GetBlockHash() != block.GetHash());
     }
-    m_node.mempool->clear();
+    BOOST_CHECK_EQUAL(m_node.mempool->size(), 1U);
+    WITH_LOCK(m_node.mempool->cs, m_node.mempool->removeRecursive(CTransaction{spends[0]}, MemPoolRemovalReason::CONFLICT));
```

Comment:
Question: is there a reason you are calling `removeRecursive()` instead of the approach from #25073 / https://github.com/bitcoin/bitcoin/pull/19909#issuecomment-1112975773 of instantiating a new mempool between test cases?

Reply:
25073 is using a mempool local to the test only (and passing it to the miner). However, here the "global" mempool is used, which is also used by validation. So creating a new mempool would also require updating the pointer in validation.

I don't think this is worth it, but I am happy to work on this, if reviewers want me to.